src/DotVVM.Framework/Routing/DotvvmRouteTable.cs

```diff
@@ -104,6 +104,21 @@ public void Add(string routeName, string url, string virtualPath, object default
             Add(group?.RouteNamePrefix + routeName, new DotvvmRoute(group?.UrlPrefix + url, group?.VirtualPathPrefix + virtualPath, defaultValues, presenterFactory, configuration));
         }
 
+        /// <summary>
+        /// Adds the specified route name.
+        /// </summary>
+        /// <typeparam name="T">Type of presenter</typeparam>
+        /// <param name="routeName">Name of the route.</param>
+        /// <param name="url">The URL.</param>
+        /// <param name="virtualPath">The virtual path of the Dothtml file.</param>
+        /// <param name="defaultValues">The default values.</param>
+        public void Add<T>(string routeName, string url, string virtualPath, object defaultValues = null) where T : IDotvvmPresenter
```

Comment:
The `virtualPath` parameter is completely redundant when the presenter is specified.

Reply:
As I told you in #463, we wanted a method with `Func<IServiceProvider, IDotvvmFactory>` parameters, this overload should be just a sugar for that (-> I'd make that an extension method). And I actually think that we don't need this overload, the presenterFactory can be quite easily used for this, see the discussion in #463

Reply:
Why dont you like generic type?